### PR TITLE
Update session management to use end-user ID + types

### PR DESCRIPTION
**Pull request by colinrotherham**
This PR adds server methods for `session.get()`, `session.set()` and `session.drop()`

I've also typed `request.auth.user` as Hapi expects it to be with all utility methods now typed

Sessions are keyed using `claims.sub` so an expired token can seamlessly re-authenticate without session loss

```js
export function createUser(credentials) {
  const claims = getUserClaims(credentials)

  if (!claims) {
    return
  }

  return /** @satisfies {UserCredentials} */ ({
    id: claims.sub,
    email: claims.email ?? '',
    displayName: claims.name ?? '',
    issuedAt: DateTime.fromSeconds(claims.iat).toUTC().toISO(),
    expiresAt: DateTime.fromSeconds(claims.exp).toUTC().toISO()
  })
}
```

**Note:** Session expiry checks are currently turned off

### CODE REVIEW

File: designer/server/src/common/helpers/auth/azure-oidc.js [hunk 1]
```diff
@@ -1,121 +1,140 @@
...
-        clientId: config.azureClientId,
-        clientSecret: config.azureClientSecret,
-        providerParams: {
-          redirect_uri: authCallbackUrl
-        },
-        cookie: 'bell-azure-oidc',
-        isSecure: false,
-        config: {
-          tenant: config.azureTenantId
-        }
```
Comment (colinrotherham) [inline designer/server/src/common/helpers/auth/azure-oidc.js:65]: This wasn't used (it seems) as it's not in the config

File: designer/server/src/common/helpers/auth/azure-oidc.js [hunk 2]
```diff
@@ -1,121 +1,140 @@
...
     password: 'testing', // 'secret'
-    name: 'Joe Bloggs',
-    id: '2133d32a'
+    user: {
+      id: 'dummy-id',
+      email: 'dummy@defra.gov.uk',
+      displayName: 'John Smith',
+      issuedAt: DateTime.now().minus({ minutes: 30 }).toUTC().toISO(),
+      expiresAt: DateTime.now().plus({ minutes: 30 }).toUTC().toISO()
+    }
```
Comment (colinrotherham) [inline designer/server/src/common/helpers/auth/azure-oidc.js:89]: All users (test fixtures, basic auth logins, OpenID Connect) now typed with Hapi `UserCredentials`

File: designer/server/src/common/helpers/auth/get-user-session.js [hunk 3]
```diff
@@ -1,24 +1,69 @@
...
+  let sessionId = session?.sessionId
+
+  // Fall back to OpenID Connect (OIDC) claim
+  if (!sessionId) {
+    const claims = getUserClaims(auth.credentials)
+
+    if (claims?.sub) {
+      sessionId = claims.sub
+    }
+  }
```
Comment (colinrotherham) [inline designer/server/src/common/helpers/auth/get-user-session.js:25]: When a user authenticates we don't know if they have a Redis session already

E.g. Maybe their token (or cookie) has expired but the Redis session hasn't

We try and grab their `sessionId` from a few places

**Review by alexluckett — APPROVED** [review]

**Comment by alexluckett** [comment]
Side note - refresh token support will return. Naz is adding a ticket into the backlog.

**Pull request merged**